amethyst_core/src/transform/components/local_transform.rs

```diff
@@ -74,6 +74,12 @@ impl Transform {
     pub fn matrix(&self) -> Matrix4<f32> {
         // This is a hot function, so manually implement the matrix-multiply to avoid a load of
         // unnecessary +0s.
+
+        // FIXME This operation assumes that `self.rotation` is a unit quaternion,
+        // which might not always be true. Perhaps this should be
+        // "foolproofed"? The probability that a user has altered a single
+        // component of the quaternion is small, but its possible as the
+        // rotation is a public field.
```

Comment:
This will be fixed thanks to the UnitQuaternion type in nalgebra.

Reply:
Nice. I just saw it so I figured I would comment on it :smile: